src/eigenvalue.cpp

```diff
@@ -0,0 +1,654 @@
+#endif
+
+  // Normalize single batch estimate of k
+  // TODO: This should be normalized by total_weight, not by n_particles
+  keff_reduced /= settings::n_particles;
+  simulation::k_generation.push_back(keff_reduced);
+}
+
+void synchronize_bank()
+{
+  simulation::time_bank.start();
+
+  // In order to properly understand the fission bank algorithm, you need to
+  // think of the fission and source bank as being one global array divided
+  // over multiple processors. At the start, each processor has a random amount
+  // of fission bank sites -- each processor needs to know the total number of
+  // sites in order to figure out the probability for selecting
+  // sites. Furthermore, each proc also needs to know where in the 'global'
+  // fission bank its own sites starts in order to ensure reproducibility by
+  // skipping ahead to the proper seed.
+
+#ifdef OPENMC_MPI
+  int64_t start = 0;
+  int64_t n_bank = simulation::fission_bank.size();
+  MPI_Exscan(&n_bank, &start, 1, MPI_INT64_T, MPI_SUM, mpi::intracomm);
+
+  // While we would expect the value of start on rank 0 to be 0, the MPI
+  // standard says that the receive buffer on rank 0 is undefined and not
+  // significant
+  if (mpi::rank == 0) start = 0;
+
+  int64_t finish = start + simulation::fission_bank.size();
+  int64_t total = finish;
+  MPI_Bcast(&total, 1, MPI_INT64_T, mpi::n_procs - 1, mpi::intracomm);
+
+#else
+  int64_t start  = 0;
+  int64_t finish = simulation::fission_bank.size();
+  int64_t total  = simulation::fission_bank.size();
+#endif
+
+  // If there are not that many particles per generation, it's possible that no
+  // fission sites were created at all on a single processor. Rather than add
+  // extra logic to treat this circumstance, we really want to ensure the user
+  // runs enough particles to avoid this in the first place.
+
+  if (simulation::fission_bank.empty()) {
+    fatal_error("No fission sites banked on MPI rank " + std::to_string(mpi::rank));
+  }
+
+  // Make sure all processors start at the same point for random sampling. Then
+  // skip ahead in the sequence using the starting index in the 'global'
+  // fission bank for each processor.
+
+  set_particle_seed(simulation::total_gen + overall_generation());
+  advance_prn_seed(start);
+
+  // Determine how many fission sites we need to sample from the source bank
+  // and the probability for selecting a site.
+
+  int64_t sites_needed;
+  if (total < settings::n_particles) {
+    sites_needed = settings::n_particles % total;
+  } else {
+    sites_needed = settings::n_particles;
+  }
+  double p_sample = static_cast<double>(sites_needed) / total;
+
+  simulation::time_bank_sample.start();
+
+  // ==========================================================================
+  // SAMPLE N_PARTICLES FROM FISSION BANK AND PLACE IN TEMP_SITES
+
+  // Allocate temporary source bank
+  int64_t index_temp = 0;
+  std::vector<Particle::Bank> temp_sites(3*simulation::work_per_rank);
```

Comment:
Question: Are you using `3` as an initial size multiplier as a "size guess" for the potential size of the expected source bank, or does "3" have other significance?  If 3 has special significance, I would note it here for clarity.